### ISSUE TITLE
How to make async/await and template strings work correctly

### ISSUE THREAD

**Issue by Richie765** [Urigo/angular2-meteor#361]
A while ago I decided to use `async` / `await` in my Meteor Angular2 projects. At first I got the error:

```
imports/portfolio/stocks/stocks.ts (150, 20): Async functions are only available when targeting ECMAScript 2015 or higher.
```

Then I put the following in my `tsconfig.json`
```JavaScript
    "target": "es6",
```

And life was good for a while. Now I needed a multi-line template string (backticks) on some client code, and I noticed some weird ``//`` characters at each line. I found out that:

```javascript
var foo = `
  multi
  line
  string
`;
```

Compiles into the following on the client (scroll to the right):
```javascript
var foo = `                                                                                                       // 127
  multi                                                                                                           //
  line                                                                                                            //
  string                                                                                                          //
`;                                                                                                                //
```

I suspect that has something to do with the `"es6"` option in my `tsconfig.json` file, because in another Blaze based project, template strings get compiled into regular strings.

How can I use `async` / `await` properly and use multi-line template strings at the same time?

**Comment by barbatus** [comment]
Read this https://github.com/barbatus/typescript#es6

**Comment by Richie765** [comment]
That doesn't solve the issue. This is about client side (Angular 2), not server side. And it also doesn't solve the multi-line template string issue (which seems to be the main issue).

**Comment by barbatus** [comment]
do you understand that es6 is not available in browsers

**Comment by Richie765** [comment]
Since version 1.2 Meteor claims ES6 support on both the server and the client.

I realize I wasn't very clear, let me rephrase the issue more clearly..

With Meteor + Typescript, ES6 isn't well supported client-side.

If I set target to ES6, async/await works, but multi-line template strings don't (I'm sure other features will break too).
If I set target to ES5, async/await doesn't work but multi-line template strings work.

I understand that to generate browser compatible code, target should be set to ES5, but then async / await won't work. That is such an essential feature these days, but to use it target must be set to ES6 and that risks browser incompatible code as I described above.

So how can I get Client-side ES6 work right with Meteor + Typescript?

**Comment by barbatus** [comment]
I just tried setting 'es6' for the todo demo.  Multi-line strings work fine in Chrome as is but yes, there is a problem when they are used as component templates. I think the issue is in the Angular2 parser not parsing them correctly. I think as well  it will be fare if you ask about that case on the Angular2 repo itself.
